src/server/dom.ts

```diff
@@ -943,10 +943,14 @@ export function waitForSelectorTask(selector: SelectorInfo, state: 'attached' |
 
       if (lastElement !== element) {
         lastElement = element;
-        if (!element)
+        if (!element) {
           progress.log(`  selector did not resolve to any element`);
-        else
+        } else {
+          const elementCount = injected.querySelectorAll(parsed, root || document).length;
```

Comment:
Please don't call querySelectorAll immediately after querySelector with the same selector - this is twice as bad. Even more, I am not sure this warning is worth the performance regression of querying a single element vs querying all of them.